src/plugins/data/server/search/aggs/aggs_service.ts

```diff
@@ -65,7 +69,10 @@ export class AggsService {
 
   public start({ fieldFormats, uiSettings, indexPatterns }: AggsStartDependencies): AggsStart {
     return {
-      asScopedToClient: async (savedObjectsClient: SavedObjectsClientContract) => {
+      asScopedToClient: async (
+        savedObjectsClient: SavedObjectsClientContract,
+        elasticsearchClient: ElasticsearchClient
+      ) => {
```

Comment:
@mattkime Just stumbled across this change now -- I'm wondering if it would be simpler to instead do:
```ts
asScoped: async (request: KibanaRequest) => {...}
```

Since both the scoped saved objects & es clients can be derived from a kibana request:
```ts
const savedObjectsClient = savedObjects.getScopedClient(req);
const elasticsearchClient = elasticsearch.client.asScoped(req);
```

The reason I called this `asScopedToClient` in the first place was because it mirrored the signature of `uiSettings.asScopedToClient` which only takes in a scoped SO client. Now that we've changed it, the naming is a bit more confusing IMHO.

In the search service (both low level & search source), we've used `asScoped(req: KibanaRequest)`, so this would also make things consistent with the other search services.

Reply:
I thought this was following the general principle of asking for for the more specific dependencies over more general. That said, I'm definitely open to improvements in the code. Mind opening a PR?

Reply:
> principle of asking for for the more specific dependencies over more general

Yeah, that's a fair point and I agree the whole `KibanaRequest` might not always be necessary. I guess it's more of a consistency-vs-explicitness tradeoff. I think I was getting hung up on the naming here in particular though; I'll give it some thought.